README.md

```diff
@@ -1 +1,284 @@
+
+```none
+#DW jobdw name=nnf-container-example-gfs2 type=gfs2 capacity=50GB
+```
+
+Then, define the container. Note the `DW_JOB_my_storage=nnf-container-example-gfs2` argument matches what is in the NNF Container Profile and maps it to the name of the GFS2 filesystem created in the DW Directive above.
+
+```none
+#DW container name=nnf-container-example profile=nnf-container-example-profile DW_JOB_my_storage=nnf-container-example-gfs2
+```
+
+## Deploy Example to Cluster
+
+**Note**: The Flux workload manager may take care of all or most of the steps in this section. This is
+the manual way of doing things. You may want to consult a Flux expert on how to drive a container
+workflow using Flux.
+
+With a working Kubernetes cluster, the previous examples can be put together and deployed on the system. The files in this repository have done that.
+
+### Create the Profile and Workflow
+
+Deploy the profile and create the workflow:
+
+```shell
+kubectl apply -f nnf-container-example.yaml
+```
+
+### Assign Nodes to the Workflow
+
+For container directives, compute nodes must be assigned to the workflow. The NNF software will trace
+the computes node back to their local NNF nodes and the container will be executed on those NNF
+nodes.
+
+For the `jobdw` directive that is included, we must define the servers (i.e. NNF nodes) and the computes.
+
+Update the `servers` and `computes` resources assigned to the workflow.
+
+**Note**: you must change
+the node names in the `allocation-*.yaml` files to match your system. `allocationCount` must match
+the number of compute nodes being targeted for that particular NNF node. In the example,
+`rabbit-node-1` is attached to `compute-node-1` and `compute-node-2`, etc.
+
+```shell
+kubectl patch --type merge --patch-file=allocation-servers.yaml servers nnf-container-example-0
+kubectl patch --type merge --patch-file=allocation-computes.yaml computes nnf-container-example
+```
+
+### Progress Workflow
+
+At this point, the workflow should be in `Proposal` state and ready:
+
+```shell
+$ kubectl get workflows
+NAME                    STATE      READY   STATUS      AGE
+nnf-container-example   Proposal   true    Completed   12m
+```
+
+Progress the workflow to the `Setup` state and wait for ready:
+
+```shell
+kubectl patch --type merge workflow nnf-container-example --patch '{"spec": {"desiredState": "Setup"}}'
+```
+
+```shell
+kubectl get workflows
+NAME                    STATE   READY   STATUS      AGE
+nnf-container-example   Setup   true    Completed   12m
+```
+
+Progress to `DataIn`:
+
+```shell
+kubectl patch --type merge workflow nnf-container-example --patch '{"spec": {"desiredState": "DataIn"}}'
+```
+
+Then `PreRun`:
+
+```shell
+kubectl patch --type merge workflow nnf-container-example --patch '{"spec": {"desiredState": "PreRun"}}'
+```
+
+The `PreRun` state will start the containers. Once the containers have started
+successfully, the state will be `Ready`. Your application is now running.
```

Comment:
...is now running on the NNF nodes that were specified in the `allocation-servers.yaml`.


Where do you explain the relationship between this container, controlled by the container profile, and the application that is running on the computes?

Reply:
> ...is now running on the NNF nodes that were specified in the allocation-servers.yaml.

*Close*. You actually specify the NNF nodes via the compute nodes that were supplied. With the given compute nodes, we trace back to their local NNF nodes and that's where the worker nodes run. I get into this in `### Assign Nodes to the Workflow`, but I will reiterate here.

> Where do you explain the relationship between this container, controlled by the container profile, and the application that is running on the computes?

That will come a bit later once I get the Port Reservation stuff working. The idea is that the application running on the comptue nodes can contact the container running on the NNF node by using an assigned Port number, which will be communicated via environment variables on the compute node. We don't currently plan to support the container hitting the compute node application.

I think I will add a section with a big fat #TODO in it just to point it out.